src/misc/json.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "3rdparty/json11/json11.hpp"
+#include "strings_func.h"
+#include "fileio_type.h"
+
+/**
+ * Parse a JSON file stored in _file_to_saveload.
+ * @param &detailed_err The detailed error message, if any, generated by the parser.
+ * @returns The file parsed as a json11:Json object.
+ */
+json11::Json ParseJsonFromLoadedFile(std::string &detailed_err, Subdirectory subdirectory)
+{
+	/* Load the JSON file as a string initially. We'll parse it soon. */
+	size_t filesize;
+	char *text{};
+	FILE *handle = FioFOpenFile(_file_to_saveload.name, "rb", subdirectory, &filesize);
+	if (handle == nullptr) return false;
+
+	text = ReallocT(text, filesize);
+	size_t read = fread(text, 1, filesize, handle);
+	fclose(handle);
+
+	if (read != filesize) return false;
+
+	/* Add a trailing \0 to mark the end of the string */
+	text = ReallocT(text, filesize + 1);
+	text[filesize] = '\0';
+
+	/* The parser doesn't like tabs or various newline characters, so we'll replace them with a space. */
+	for (char *p = text; *p != '\0'; p++) {
+		if (*p == '\t' || *p == '\r' || *p == '\n' || *p == '\r\n') *p = ' ';
```

Comment:
As reported by CI '\r\n' is indeed not a single char but 2 chars, and they are already handled anyway :)